src/client/debugger/jupyter/debugControllers.ts

```diff
@@ -107,8 +107,12 @@ export class RunByLineController implements IDebuggingDelegate {
         // start the process of updating the variables view.
         const stResponse = await this.debugAdapter.stackTrace({ threadId, startFrame: 0, levels: 1 });
 
-        const sf = stResponse.stackFrames[0];
-        return !!sf.source && sf.source.path !== this.debugCell.document.uri.toString();
+        if (stResponse && stResponse.stackFrames[0]) {
+            const sf = stResponse.stackFrames[0];
+            return !!sf.source && sf.source.path !== this.debugCell.document.uri.toString();
+        }
+
+        return false;
```

Comment:
@roblourens my paranoid side thinks this logic might be backwards, and we might start getting random situations where RBL stops in other files.
If we say true on this cases, we stepIn on stop, and might randomly step 2 lines.

IDK which one is better,